experimental/generation/generator/packages/library/templates/generator.lg

```diff
@@ -264,6 +264,53 @@
 }
 ```
 
+# addArrayDialogWithChecking(entity)
+-```
+{
+    "$schema": "${appSchema}",
+    "$kind": "Microsoft.OnAssignEntity",
+    "operation": "Add()",
+    "property": "${property}",
+    "entity": "${entity}",
+    "actions": [
+        {
+            "$kind": "Microsoft.IfCondition",
+            "condition": "!contains($${property}, @${entity})",
+            "actions": [
+                {
+                    "$kind": "Microsoft.SendActivity",
+                    "activity": "\${confirmationText('${property}',  @${entity})}"
```

Comment:
>'${property}' [](start = 53, length = 13)

This needs to be propertyName(property) evaluated at generation time.  If property = 'kindsOfMeat' that is not we want in output.  If at generation time you evaluate ${propertyName(property)} it will become 'kinds of meat'.